km-core/src/main/java/com/xiaojukeji/know/streaming/km/core/cache/ZookeeperLocalCache.java

```diff
@@ -0,0 +1,46 @@
+package com.xiaojukeji.know.streaming.km.core.cache;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.xiaojukeji.know.streaming.km.common.bean.entity.zookeeper.fourletterword.BaseFourLetterWordCmdData;
+
+import java.util.concurrent.TimeUnit;
+
+public class ZookeeperLocalCache {
+    private static final Cache<String, String> fourLetterCmdFailedServerCache = Caffeine.newBuilder()
+            .expireAfterWrite(10, TimeUnit.MINUTES)
+            .maximumSize(10000)
+            .build();
+
+    private static final Cache<String, BaseFourLetterWordCmdData> fourLetterCmdDataCache = Caffeine.newBuilder()
+            .expireAfterWrite(60, TimeUnit.SECONDS)
+            .maximumSize(10000)
+            .build();
+
+    public static boolean canUse(String host, int port, String cmd) {
+        String data = fourLetterCmdFailedServerCache.getIfPresent(gen4lwFailedKey(host, port, cmd));
+
+        return data == null;
+    }
+
+    public static void setFailed(String host, int port, String cmd) {
+        fourLetterCmdFailedServerCache.put(gen4lwFailedKey(host, port, cmd), "");
```

Comment:
The value is ""(empty value) here, however, the method `canUse` return `data == null`. It should be return `data == null || "".equals(data)`, right?

Reply:
```
    // the data in fourLetterCmdFailedServerCache will expire after write 10 minutes, and if the data exist in cache 
    // means still can not use the (host, port, cmd).
    // And if the fourLetterCmdFailedServerCache's data expired, the getIfPresent method will return null.
    private static final Cache<String, String> fourLetterCmdFailedServerCache = Caffeine.newBuilder()
            .expireAfterWrite(10, TimeUnit.MINUTES)
            .maximumSize(10000)
            .build();

    public static boolean canUse(String host, int port, String cmd) {
        String data = fourLetterCmdFailedServerCache.getIfPresent(gen4lwFailedKey(host, port, cmd));

        return data == null;
    }
```